sklearn/metrics/_classification.py

```diff
@@ -283,6 +283,10 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None,
 
     check_consistent_length(y_true, y_pred, sample_weight)
 
+    if normalize not in {'true', 'pred', 'all', None}:
```

Comment:
```suggestion
    if normalize not in ['true', 'pred', 'all', None]:
```
as otherwise you would get a type error if `normalize` is unhashable,
```
>>> {'a': 2} in {'true', 'false'}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'dict'
>>> >>> {'a': 2} in ['true', 'false']
False
```